server/task_executor_database_create.go

```diff
@@ -120,6 +120,44 @@ func (exec *DatabaseCreateTaskExecutor) RunOnce(ctx context.Context, server *Ser
 		return true, nil, err
 	}
 
+	// After the database creation statement executed rightly,
+	// insert its information to our database.
+	databaseCreate := &api.DatabaseCreate{
+		CreatorID:     api.SystemBotID,
+		ProjectID:     payload.ProjectID,
+		InstanceID:    task.InstanceID,
+		EnvironmentID: instance.EnvironmentID,
+		Name:          payload.DatabaseName,
+		CharacterSet:  payload.CharacterSet,
+		Collation:     payload.Collation,
+		Labels:        &payload.Labels,
+		SchemaVersion: payload.SchemaVersion,
+	}
+	database, err := server.DatabaseService.CreateDatabase(ctx, databaseCreate)
+	if err == nil {
+		// Update task database_id with the created database id
+		taskDatabaseIDPatch := &api.TaskPatch{
+			ID:         task.ID,
+			UpdaterID:  api.SystemBotID,
+			DatabaseID: &database.ID,
+		}
+		if _, err = server.TaskService.PatchTask(ctx, taskDatabaseIDPatch); err != nil {
```

Comment:
In this case, we may want to fail-close on errors (instead of ignoring errors currently here) upon creating database row and updating task.